Fix: Mapilio Bubbles

keiffer213 opened the pull request:
## Summary & Changes 📃
- **Resolves:** #10898 

- **Summary:** (Briefly describe what this PR does)
  - This PR fixes the issue of the Mapilio bubbles being inconsistent with the other map data overlays such as Mapillary, KartaView, and Panoramax.
  - The expected behavior is that when you are editing in OpenStreetMap and are between the zoom level of 10-16, only the street lines (blue) for Mapilio display on the map. Once the zoom level is at 16 or above, then the Mapillio bubbles will show up. When selecting a bubble, an image view should display at the bottom left of the screen. Once, you zoom out with a zoom value < 16, the bubble that selected (yellow )will disappear as well as the image viewer.Also, when zoom <16, the bubble selected (yellow) get reset and the image that selected will disappear. When turn off mapilio, the bubble selected earlier (yellow ) will get reset and image will disappear.

- **Changes:**
  - Separated "minZoom" to "imageMineZoom" and "lineMinZoom" (following the format of panoramax_images.js)
  - Refactored the code to have similar functionality with [panoramax_images.js](https://github.com/openstreetmap/iD/blob/develop/modules/svg/panoramax_images.js)
  - Fixed the conditional statements for when lines and bubbles render
  


## Screenshots / Visual Aids 🔎
<sub><i>📌 **Required for:** UI changes, layout updates, or bug fixes.</i></sub>

<details>
  <summary> Expand ⬇️ </summary>
  <!-- add GIFs/Screenshots/Videos/Diagrams here -->
  
cant_tick_when_zoom_small_than_10

![cant_tick_when_zoom_small_than_10](https://github.com/user-attachments/assets/257a7194-94b8-4b65-9905-baf3d109dcaf)

could_tick_and_show_line
![could_tick_and_show_line](https://github.com/user-attachments/assets/92e0a283-107e-4b51-b78a-ab46bd5f6be4)

![show_line](https://github.com/user-attachments/assets/bcb1cbd4-e93d-413b-995c-6cbdbdf13af0)


Video showing how bubble work

https://github.com/user-attachments/assets/23e93754-a896-425e-b55a-36873e3cc8d7


</details>


## How to Test 🧪
1. **Steps to Reproduce:**
   - Step 1: Run this [branch](https://github.com/keiffer213/iD/tree/mapilio_bubbles)
   - Step 2: On the edit map, enable mapilio bubbles and observe the behavior.

2. **Expected Behavior:** (Describe what should happen):
- Cannot select mapilio if zoom <10. If at zoom=10, tick on mapilio will appear the mapilio line.
- if zoom >= 16, the mapilio bubbles will apear and if zoom >=18, the bubbles will show fieldzoom.
- When one bubble is selected, it will turn to yellow and its size will be increased. The bubbles that selected will be adjust its size base on the zoom whose scale is  > 16. If zoom <16, all bubbles will be disappeared include yellow bubbles. When zoom back >=16, the bubble will be unselected and turn to blue like the others bubbles.


## Checklist ✅

- [x] I have **tested** this PR **locally** and it works as expected.
- [x] This PR **resolves an issue** (`Resolves #issue-number`).

keiffer213 committed:
Update mapilio_images.js

keiffer213 committed:
Update mapilio_images.js

fix min zoom

keiffer213 committed:
Update mapilio_images.js

keiffer213 committed:
Update mapilio_images.js

keiffer213 committed:
Update mapilio_images.js

keiffer213 committed:
Update mapilio_images.js

uttamalld reviewed (commented):
changes looks good. Please add screenshots of the behavior after these changes highlighting the zoom value to make it easier for review to understand when opening a PR against develop branch.

haipq07 committed:
upload bubble part

keiffer213 committed:
Update mapilio_images.js

Added semicolons at the end and declared service variable in the drawImages.enabled function

tyrasd commented on modules/svg/mapilio_images.js:
```diff
@@ -7,30 +7,107 @@ import {svgPath, svgPointTransform} from './helpers';
+                const tempIds = await service.getUserIds(username);
+
+                _activeIds = {};
+                tempIds.forEach(id => _activeIds[id] = true);
```
when a username filter is set, this crashes:

![image](https://github.com/user-attachments/assets/142cf5f2-bb9e-4861-8fa5-75147a759f50)



I'm not sure if the mapilio API officially supports user information in the public API. I found the endpoint `https://end.mapilio.com/api/search-user?options[parameters][keywords]=…` which kind of works, but a) the search is fuzzy and b) the endpoint is not included in the API documentation at https://docs.mapilio.com/

tyrasd commented on modules/svg/mapilio_images.js:
```diff
@@ -254,31 +299,47 @@ export function svgMapilioImages(projection, context, dispatch) {
             }
         } else {
             dispatch.call('photoDatesChanged', this, 'mapilio', []);
+            service.hideViewer(context);
```
What's the intention here? I think the previous behavior was the intended one: keep the viewer open in case the user just wants to temporarily hide the sequences and photo markers.

tyrasd reviewed (changes requested):
Hey, thanks for this fix. It is indeed intended to show the photo markers only starting from relatively a high zoom level. I was wondering about the username-filtering code (see comment below). And there's a failing test (also, see below). Otherwise this looks fine.

PS: If you like, can you please also add a short line about this to the repo's `CHANGELOG.md` file? Feel free to also link your user profile by appending `…, thanks [@keiffer213])`

keiffer213 committed:
Merge branch 'develop' into mapilio_bubbles

keiffer213 committed:
Update CHANGELOG.md

Added PR and developers in CHANGELOG

keiffer213 committed:
Update mapilio_images.js

mapilio api doesn't support users currently and removed service.hideViewer and SelectImage code sections

keiffer213 commented:
> Hey, thanks for this fix. It is indeed intended to show the photo markers only starting from relatively a high zoom level. I was wondering about the username-filtering code (see comment below). And there's a failing test (also, see below). Otherwise this looks fine.
> 
> PS: If you like, can you please also add a short line about this to the repo's `CHANGELOG.md` file? Feel free to also link your user profile by appending `…, thanks [@keiffer213])`

Thank you for the feedback! I have added myself and @haipq07 to the CHANGELOG.md along with adding this PR

keiffer213 committed:
Merge pull request #2 from openstreetmap/develop

merge

tyrasd committed:
add changelog entry

tyrasd committed:
don't hide viewer on zoom events
